File: backend/agents/resumeReviewerAgent.js
const { callGroq } = require('../services/groqService');
const { parseJSON, normalizeResumeReview } = require('../utils/responseParser');
const { RAGPipeline } = require('../rag/ragPipeline');

/**
 * AGENT 1: Resume Reviewer Agent
 *
 * Analyzes resume quality against the job description.
 * Returns: resumeScore, strengths, weaknesses, improvements
 */
const runResumeReviewerAgent = async (ragPipeline) => {
  console.log('\n🤖 Agent 1: Resume Reviewer running...');

  // Retrieve relevant chunks from both resume and JD
  const resumeContext = await ragPipeline.retrieve(
    'professional experience skills projects achievements education',
    6,
    'resume'
  );
  const jdContext = await ragPipeline.retrieve(
    'required qualifications skills experience responsibilities',
    4,
    'jd'
  );

  const systemPrompt = `You are an expert Resume Reviewer with 15+ years of experience in HR and talent acquisition for top tech companies. You provide honest, actionable feedback.`;

  const userPrompt = `Analyze this resume against the job description.

RESUME CONTENT:
${RAGPipeline.truncate(resumeContext, 2500)}

JOB DESCRIPTION:
${RAGPipeline.truncate(jdContext, 1500)}

Evaluate:
1. Content quality and job relevance
2. Achievement quantification (numbers, percentages, impact)
3. Keyword optimization for the role
4. Structural clarity and professionalism
5. Overall marketability

Return ONLY a JSON object:
{
  "resumeScore": <0-100>,
  "strengths": ["strength1", "strength2", "strength3", "strength4"],
  "weaknesses": ["weakness1", "weakness2", "weakness3"],
  "improvements": ["specific improvement1", "specific improvement2", "specific improvement3", "specific improvement4"]
}`;

  const raw = await callGroq(systemPrompt, userPrompt, { temperature: 0.2 });
  const parsed = parseJSON(raw, {});
  return normalizeResumeReview(parsed);
};

module.exports = { runResumeReviewerAgent };

File: ai-service/agents/resume_reviewer_agent.py
"""
Agent 1: Resume Reviewer Agent
================================
Analyzes resume quality, strengths, weaknesses, and provides a score.

Output:
    {
        "resumeScore": 82,
        "strengths": [...],
        "weaknesses": [...],
        "improvements": [...]
    }
"""

from rag.rag_pipeline import RAGPipeline
from services.groq_service import call_groq
from utils.response_parser import parse_json, normalize_resume_review
from prompts import resume_reviewer_prompt


async def run_resume_reviewer_agent(rag: RAGPipeline) -> dict:
    """
    AGENT 1: Resume Reviewer

    Uses RAG to retrieve the most relevant resume and JD sections,
    then asks the LLM to score and analyze the resume quality.

    Args:
        rag: Initialized RAGPipeline instance

    Returns:
        Normalized resume review dict with score, strengths, weaknesses, improvements
    """
    print("\n🤖 Agent 1: Resume Reviewer running...")

    # ─── RAG Retrieval ────────────────────────────────────────
    resume_context = await rag.retrieve(
        "professional experience skills projects achievements education certifications",
        top_k=6,
        source="resume",
    )
    jd_context = await rag.retrieve(
        "required qualifications skills experience responsibilities must have",
        top_k=4,
        source="jd",
    )

    # ─── Prompt Construction ──────────────────────────────────
    system_prompt = resume_reviewer_prompt.get_system_prompt()
    user_prompt = resume_reviewer_prompt.get_user_prompt(
        resume_context=RAGPipeline.truncate(resume_context, 2500),
        jd_context=RAGPipeline.truncate(jd_context, 1500),
    )

    # ─── LLM Call ─────────────────────────────────────────────
    raw = await call_groq(system_prompt, user_prompt, temperature=0.2)
    parsed = parse_json(raw, fallback={})
    result = normalize_resume_review(parsed)

    print(f"   ✅ Resume Score: {result['resumeScore']}/100")
    return result
